packages/key-management/src/util/ownSignatureKeyPaths.ts

```diff
@@ -82,5 +116,9 @@ export const ownSignatureKeyPaths = async (
     ).filter(isNotNil)
   ).map(({ type, index }) => ({ index, role: Number(type) }));
 
-  return [...paymentKeyPaths, ...getStakingKeyPaths(knownAddresses, txBody)];
+  return [
```

Comment:
Might be already returning required signers key paths as payment or stake keys
```suggestion
  return uniqBy([
```